application_form/services/lottery/utils.py

```diff
@@ -19,3 +26,18 @@ def _save_application_order(apartment: Apartment) -> None:
             application_apartment=apartment_reservation.application_apartment,
             result_position=apartment_reservation.queue_position,
         )
+
+
+def _validate_project_has_applications(project: Project):
+    application_count = Application.objects.filter(apartments__project=project).count()
+    if application_count == 0:
+        raise ProjectDoesNotHaveApplicationsException()
+
+
+def _validate_project_application_time_has_finished(project: Project):
+    elastic_object = project.get_elastic_document()
+    if (
+        not elastic_object.project_application_end_time
+        or elastic_object.project_application_end_time.date() > date.today()
```

Comment:
Hmm, so this means that if the application end date == today, the application is already finished? Is that correct behavior?

Reply:
Good found! Here was two problems: End time is DateTime and it should also check ==.